### DISCUSSION TITLE
Missing documentation

### DISCUSSION THREAD

**Discussion by tobias-tengler** [ChilliCream/hotchocolate#5073] [Feature Request]
The following is a list of features / functionality that still needs to be documented in Hot Chocolate.

**Missing**

- [ ] Automatic type registration
- [ ] Multiple schemas
- [ ] Type interceptors
- [ ] Error handling
- [ ] Request options
- [ ] Schema options
- [ ] Optional / DefaultValue
- [ ] Queryplan / Execution strategies (serial vs. parallel)
- [ ] Conventions
(for naming conventions, cross-link to note added in [this PR](https://github.com/ChilliCream/hotchocolate/pull/4661) about use with xml docs)
- [ ] Resolver Types (pure vs async)
- [ ] Schema-first APIs
- [ ] Coding approaches (pros & cons / some examples)
- [ ] Subscription authentication (currently only a workaround exists)
- [ ] Testing
- [ ] DataLoaderOptions
- [ ] UnsafeCreate
- [ ] Introspection Extensions
- [ ] Directive Introspection
- [ ] Deployment (AZF & AWS Lambda)
- [ ] Subscription cleanup / switching between apollo / graphql-ws
- [ ] Client controlled nullability
- [ ] Oneof
- [ ] Stream / Defer (Incremental delivery)

**Improvements**

- [ ] Describe possible solutions for input polymorphism
- [ ] Data extension methods like .Sort(context)
- [ ] More details about Unions and input object types
- [ ] Add interface and enum type extensions
- [ ] Showcase query complexity with Redis
- [ ] (Optional) Application Insights Integration

**v13**

- [ ] New relay getbyid feature
- [ ] New way to create data loaders
- [ ] Change EF integration recommendation to Resolver-scoped

**Need to be reviewed**

- [ ] DataLoader
- [ ] Filtering
- [ ] Sorting
- [ ] MongoDB
- [ ] Neo4J
- [ ] Spatial
- [ ] Automatic persisted queries
- [ ] Operation complexity

This list is not exhaustive, feel free to edit the task list with missing pieces.

**Comment by michaelstaib** [comment]
**Dynamic Schemas**

- [ ] Type Modules -> Added support for type modules.
- [ ] UnsafeCreate
- [ ] TypeInterceptors

**Comment by michaelstaib** [comment]
Query Plan

- [ ] Resolver Types (pure vs async)
- [ ]  Control behavior or plan serial/parallel

**Comment by michaelstaib** [comment]
- [ ] Resolver Compiler

**Comment by michaelstaib** [comment]
- [x] Introspection Authorization
- [ ] Introspection Extensions
- [ ] Directive Introspection

**Comment by glen-84** [comment]
Why is this a discussion?

**Reply by michaelstaib** [reply]
We are keeping a lot of these things discussions. Also design proposals are discussions.

**Comment by glen-84** [comment]
Value formatters can also be added to the (long) list.
